firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift

```diff
@@ -1697,9 +1697,17 @@ class BrowserViewController: UIViewController,
             // Ensure that the tab title *actually* changed to prevent repeated calls
             // to navigateInTab(tab:).
             guard let title = tab.title else { break }
-            if !title.isEmpty && title != tab.lastTitle {
-                tab.lastTitle = title
-                navigateInTab(tab: tab, webViewStatus: .title)
+            if !title.isEmpty {
+                if title != tab.lastTitle {
+                    tab.lastTitle = title
+                    navigateInTab(tab: tab, webViewStatus: .title)
+                } else {
+                    if let readerMode = tab.getContentScript(name: ReaderMode.name()) as? ReaderMode {
+                        if readerMode.state == .active {
+                            navigateInTab(tab: tab, webViewStatus: .title)
+                        }
+                    }
+                }
             }
             TelemetryWrapper.recordEvent(category: .action, method: .navigate, object: .tab)
```

Comment:
this seems like it should be called when calling the navigate tab, but separate from this ticket so we can leave as is.